### PR TITLE
Added functionality to view yaml files inline with browser

### PR DESCRIPTION
**Pull request by gbyrketosc**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201625178070145/1201628271829112) by [Unito](https://www.unito.io)

### CODE REVIEW

File: apps/dashboard/app/models/files.rb [hunk 1]
```diff
@@ -132,12 +132,24 @@ def self.mime_type_for_preview(type)
...
-    end
-  end
+
+    # NOTE:  due to how most modern browsers are handling the yaml mime-types,
+    # we have to cast the yaml mime-types to text/plain mime-type so we can
+    # view the yaml files inline with the browser.
+    # Would like to have a pattern match for the  yaml mime-types; something like application/*yml and application/*yaml
+    # but couldn't find one that worked. Therefore, for now we're using full mime-type names for yaml mime-types.
+
+    castToTextArray = ['text/*','application/x-yml','application/x-yaml','application/yml','application/yaml']
```
Comment (johrstrom) [inline apps/dashboard/app/models/files.rb:142]: ruby variables are `snake_case`.

File: apps/dashboard/app/models/files.rb [hunk 2]
```diff
@@ -132,12 +132,24 @@ def self.mime_type_for_preview(type)
...
+
+    castToTextArray.each { |mime| 
+      if ! %r{#{mime}}.match(type).nil?
+        type = 'text/plain; charset=utf-8'
+      end
+    }
+
+    type
```
Comment (johrstrom) [inline apps/dashboard/app/models/files.rb:150]: For multi-line blocks you want to you want to use the `do end` type notation. 

```ruby
foo.each do |ele|
  # the block
end
```
But I got a one liner for ya

```suggestion
    type = 'text/plain; charset=utf-8' if plain_cast_array.any? { |mime| /#{mime}/.match?(type) }

    type
```

I'll have to check the rubocop `Style/RegexpLiteral` - but looking at it now I think I do prefer slashes.

File: apps/dashboard/app/models/files.rb (hunk 1 above)
Comment (gbyrketosc) [reply apps/dashboard/app/models/files.rb]: changed.

**Review by johrstrom — CHANGES_REQUESTED** [review]
code itself seems fine, but you have some linting errors.

**Review by johrstrom — APPROVED** [review]

**Pull request merged**